docs/src/operate/c8y/supported-operations.md

```diff
@@ -140,34 +140,36 @@ The `Supported Operations API` can also be used to add support for custom operat
 For e.g: Cumulocity supports the `c8y_Command` that enables execution of shell commands on the device from the cloud.
 The same operation files can be used to define how the mapper should handle these operations when triggered from the cloud.
 
-### Adding new custom operation
+### Creating new custom operation
+
+Custom operations can be delivered by SmartRest message or C8Y JSON depending on the content of the operation file. The former is limited to SmartRest input. The latter moves the actual execution of the operation to `tedge-agent` where is executed by the workflow.
+
+:::note
+The operation file needs to be readable by %%te%% user - `tedge` - and should have permissions `644`.
+The filename **MUST** only use alphanumeric and underscore characters, e.g. `A-Z`, `a-z`, `0-9` and `_`. Alternatively, it can have `.template` extension for a master custom operation definition file.
+You cannot use a dash "-", or any other characters in the filename, otherwise the custom operation definition will be ignored.
+:::
+
+#### Create operation delivered via SmartRest message
 
 An operation file must be created with the name of the operation:
 
 ```sh
 sudo -u tedge touch /etc/tedge/operations/c8y/c8y_Command
 ```
 
-...with the following content:
+... with the definition containing `on_message` field:
 
 ```toml title="file: /etc/tedge/operations/c8y/c8y_Command"
 [exec]
-  topic = "c8y/s/ds"
-  on_message = "511"
-  command = "/etc/tedge/operations/command"
-  timeout = 10
+topic = "c8y/s/ds"
+on_message = "511"
+command = "/etc/tedge/operations/command"
+timeout = 10
 ```
 
-In this example, the mapper is configured to pick up the `c8y_Command` operation message received on `c8y/s/ds` topic
-with the SmartRest template message prefix `511`.
-When such a message is received, the operation plugin located at `/etc/tedge/operations/command` would be executed.
-The operation is configured to `timeout` after 10 seconds, to avoid it from running for too long/forever.
+The mapper is configured to pick up the `c8y_Command` operation message received on `c8y/s/ds` topic with the SmartRest template message prefix `511`. When such a message is received, the operation plugin located at `/etc/tedge/operations/command` would be executed. The operation is configured to `timeout` after 10 seconds, to avoid it from running for too long/forever.
 
-:::note
-The operation file needs to be readable by %%te%% user - `tedge` - and should have permissions `644`.
-The filename **MUST** only use alphanumeric and underscore characters, e.g. `A-Z`, `a-z`, `0-9` and `_`.
-You cannot use a dash "-", or any other characters in the filename, otherwise the custom operation definition will be ignored.
-:::
 
 :::note
 The `timeout` that is configured will be in seconds.
```

Comment:
I would move this note related to timeouts in the sub-section that details the operation configuration parameters

Reply:
Resolved by a267f04581ec93b1644543df4d45e998b590aaa8